File: frontend/src/types/match.ts
export type ViewerRole = 'SCORER' | 'VIEWER';

export type MatchStatus =
  | 'INNINGS_1'
  | 'INNINGS_BREAK'
  | 'INNINGS_2'
  | 'COMPLETED';

export type EventType = 'RUN' | 'WICKET' | 'WIDE' | 'NO_BALL';

export interface InningsScore {
  score: number;
  wickets: number;
  overs: string;
}

export interface MatchResult {
  winner: string | null;
  type: 'RUNS' | 'CHASE' | 'TIE' | string;
  margin: number;
}

export interface Match {
  match_id: number;
  team_1: string;
  team_2: string;
  overs_per_innings: number;
  current_innings: number;
  status: MatchStatus;
  scorer_active: boolean;
  viewer_role: ViewerRole;
  innings_1: InningsScore;
  innings_2: InningsScore;
  result: MatchResult | null;
  recent_events?: string[];
}

export interface MatchCreateRequest {
  team_1: string;
  team_2: string;
  overs: number;
}

export interface MatchCreateResponse {
  match_id: number;
  status: string;
}

export interface ScoringEventRequest {
  type: EventType;
  runs?: number;
}

File: backend/app/schemas/match.py
from pydantic import BaseModel, Field
from typing import Literal

class MatchCreate(BaseModel):
    team_1 : str = Field(min_length=1, max_length=100) 
    team_2 : str = Field(min_length=1, max_length=100) 
    overs : int = Field(gt=0, le=100)

class MatchCreateResponse(BaseModel):
    match_id : int
    status : str

class InningsScoreResponse(BaseModel):
    score: int
    wickets: int
    overs: str

class MatchResultResponse(BaseModel):
    winner: str | None
    type: str
    margin: int

class MatchResponse(BaseModel):
    match_id: int
    team_1: str
    team_2: str
    overs_per_innings: int
    current_innings: int
    status: str
    scorer_active: bool
    viewer_role: str
    innings_1: InningsScoreResponse
    innings_2: InningsScoreResponse
    result : MatchResultResponse | None
    recent_events: list[str] = Field(default_factory=list)

class ScoringEventCreate(BaseModel):
    type : Literal["RUN", "WICKET", "NO_BALL", "WIDE"]
    runs : int | None = None

